test/utils/mockutils/server.go

```diff
@@ -5,11 +5,18 @@ import (
 	"net/http"
 )
 
-func RespondWithMethodExpectation(w http.ResponseWriter, r *http.Request, methodName string) {
+func RespondNoContentForMethod(w http.ResponseWriter, r *http.Request, methodName string) {
+	RespondToMethod(w, r, methodName, func() {
+		w.WriteHeader(http.StatusNoContent)
+	})
+}
+
+func RespondToMethod(w http.ResponseWriter, r *http.Request, methodName string, onSuccess func()) {
 	// if method is not as expected we return error code so the test will fail
 	// and with response payload which will be a helpful message for debugging
 	if r.Method == methodName {
-		w.WriteHeader(http.StatusNoContent)
+		// if method is matching execute callback
+		onSuccess()
```

Comment:
Some POST/PUT/PATCH/DELETE may get different status code, response body which is handled by callback.

We are interested in methods to match, because of the nature of Write function.